data/archiver/archiver.py

```diff
@@ -1,40 +1,45 @@
-import os
 import logging
 from data.archiver.aws_s3_client import AwsS3
 from data.archiver.ftp_uploader import FtpUploader
+from data.archiver.ingest_api import Ingest
 from data.archiver.stream import S3FTPStreamer
 from data.archiver.utils import compress, md5
 from data.archiver.dataclass import DataArchiverRequest, DataArchiverResult, FileResult
-from data.archiver.ingest_api import Ingest
 
 
 class Archiver:
 
-    def __init__(self):
-        self.ingest_api = Ingest()
-        self.aws_client = AwsS3()
+    def __init__(self, ingest_cli: Ingest, aws_cli: AwsS3):
+        self.ingest_cli = ingest_cli
+        self.aws_cli = aws_cli
         self.ftp = None
         
         self.logger = logging.getLogger(__name__)
         self.logger.setLevel(logging.INFO)
 
     def start(self, req: DataArchiverRequest):
         self.logger.info(req)
-        self.logger.info(f'Getting sequence files for submission {req.sub_uuid} from INGEST_API')
-        sequence_files = self.ingest_api.get_sequence_files(req.sub_uuid)
+        self.logger.info(f'Getting sequence files for submission {req.sub_uuid} from Ingest.')
+        sequence_files = self.ingest_cli.get_sequence_files(req.sub_uuid)
         self.logger.info(sequence_files)
 
         if not sequence_files:
-            res = DataArchiverResult(req.sub_uuid, success=False, error='No sequence files in submission')
+            res = DataArchiverResult(req.sub_uuid, success=False, error='No sequence files in submission.')
             self.logger.info(res)
             return res
 
         if not req.files:
             # archive all files
-            res_files = list(map(lambda f: FileResult(f["file_name"], f["cloud_url"]), sequence_files))
+            res_files = list(map(lambda f: FileResult(f["uuid"], f["file_name"], f["cloud_url"]), sequence_files))
```

Comment:
Not sure how you mean. New FileResult instance is being created using its constructor. Perhaps it's not very clear, needing to pass multiple args which is not ideal, but they are the mandatory args only; Python doesn't support constructor overloading. we could have a @classmethod annotated function but I don't see how it would be clearer.

```
    def __init__(self, uuid, file_name, cloud_url, size=0, compressed=False, md5=None, ena_upload_path=None, success=True, error=None):
```

Reply:
I meant adding to `FileResult` a factory method or even creating a builder class (might be overkill here):

```python
class FileResult:
    # ...
    @class_method
    def from_file(file):
        return FileResult(file["uuid"], file["file_name"], file["cloud_url"])) 
```

so that the usage does not over burden the reader with too many details in to deep abstraction level:

```python
res_files = list(map(lambda f: FileResult.from_file(f), sequence_files))
```